aws/main.tf

```diff
@@ -1,47 +1,84 @@
-data aws_eks_cluster "my_eks" {
-  count = var.modules_info.eks.create ? 0 : 1
-  name = var.modules_info.eks.cluster_id
+locals {
+  vpc_id = var.modules_info.vpc.create ? module.vpc[0].vpc_id : var.modules_info.vpc.id
+  #private_subnets = var.modules_info.vpc.create ? module.vpc[0].private_subnets_cidr_blocks : var.modules_info.vpc.private_subnets_cidr_blocks
+  efs_id                             = var.modules_info.efs.create ? module.efs.efs_id : var.modules_info.efs.id
+  cluster_certificate_authority_data = var.modules_info.eks.create ? module.eks[0].cluster_certificate_authority_data : data.aws_eks_cluster.my_eks[0].certificate_authority[0].data
+  cluster_endpoint                   = var.modules_info.eks.create ? module.eks[0].cluster_endpoint : data.aws_eks_cluster.my_eks[0].endpoint
+  private_subnets_ids                = var.modules_info.vpc.create ? module.vpc[0].private_subnets : data.aws_subnets.private[0].ids
 }
 
-locals {
-  vpc_id          = var.modules_info.vpc.create ? module.vpc[0].vpc_id : var.modules_info.vpc.id
-  private_subnets = var.modules_info.vpc.create ? module.vpc[0].private_subnets : var.modules_info.vpc.private_subnets
-  efs_id          = var.modules_info.efs.create ? module.efs[0].efs_id : var.modules_info.efs.id
+#TODO figure out why the data 
+data "aws_subnets" "private" {
+  count = var.modules_info.vpc.create ? 0 : 1
+  filter {
+    name   = "vpc-id"
+    values = [var.vpc_id]
+  }
+
+  tags = {
+    tier = "private"
+  }
+}
+
+data "aws_eks_cluster" "my_eks" {
+  count = var.modules_info.eks.create ? 0 : 1
+  name  = var.modules_info.eks.cluster_id
 }
 
+# TODO 
+# figure out why the CSI driver fails to deploy
+
 module "vpc" {
-  count           = var.modules_info.vpc.create ? 1 : 0
-  source          = "./vpc"
+  count  = var.modules_info.vpc.create ? 1 : 0
+  source = "./vpc"
 
   cluster_name    = var.cluster_name
   cidr            = var.cidr
-  private_subnets = var.private_subnets
-  public_subnets  = var.public_subnets
+  private_subnets = var.private_subnets_cidr_blocks
+  public_subnets  = var.public_subnets_cidr_blocks
 }
 
-module "eks" {
-  depends_on = [module.vpc]
```

Comment:
the depends on created errors

https://github.com/terraform-aws-modules/terraform-aws-eks/issues/2612#issuecomment-1545675279